patches/server/0966-Paper-Item-API.patch

```diff
@@ -0,0 +1,5048 @@
++
++        return HoverEvent.showItem(op.apply(HoverEvent.ShowItem.of(key, this.itemStack.getCount(), PaperAdventure.asBinaryTagHolder(this.itemStack.getTag()))));
++    }
++
++    @Override
++    public @NotNull String translationKey() {
++        return null;
++    }
++}
+diff --git a/src/main/java/io/papermc/paper/item/meta/MetaAxolotlBucket.java b/src/main/java/io/papermc/paper/item/meta/MetaAxolotlBucket.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..36a56545bf29738abff5d02c84a52c9a5e7cc3dd
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/item/meta/MetaAxolotlBucket.java
+@@ -0,0 +1,43 @@
++package io.papermc.paper.item.meta;
++
++import io.papermc.paper.inventory.item.properties.ItemProperties;
++import io.papermc.paper.property.PropertyHolder;
++import io.papermc.paper.property.nbt.NbtPropertyHolder;
++import net.minecraft.nbt.CompoundTag;
++import org.bukkit.Material;
++import org.bukkit.entity.Axolotl;
++import org.bukkit.inventory.meta.AxolotlBucketMeta;
++import org.jetbrains.annotations.NotNull;
++
++public class MetaAxolotlBucket extends PaperMeta implements AxolotlBucketMeta {
++
++    public MetaAxolotlBucket(PropertyHolder.Mutable mutable, Material material) {
++        super(mutable, material);
++    }
++
++    @Override
++    public @NotNull Axolotl.Variant getVariant() {
++        return this.propertyHolder.get(ItemProperties.VARIANT);
++    }
++
++    @Override
++    public void setVariant(Axolotl.@NotNull Variant variant) {
++        this.propertyHolder.set(ItemProperties.VARIANT, variant);
++    }
++
++    @Override
++    public boolean hasVariant() {
++        return this.propertyHolder.hasProperty(ItemProperties.VARIANT);
++    }
++
++    @Override
++    public @NotNull MetaAxolotlBucket clone() {
++        return new MetaAxolotlBucket(this.propertyHolder.clone(), material);
++    }
++
++    @Override
++    public void migrateFrom(PropertyHolder.Mutable mutable) {
++        this.propertyHolder.unset(ItemProperties.VARIANT);
++        this.propertyHolder.unset(ItemProperties.ENTITY_TAG);
++    }
++}
+diff --git a/src/main/java/io/papermc/paper/item/meta/MetaBanner.java b/src/main/java/io/papermc/paper/item/meta/MetaBanner.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..0f694872100e726def29548e62be84f24e263b8b
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/item/meta/MetaBanner.java
+@@ -0,0 +1,87 @@
++package io.papermc.paper.item.meta;
++
++import io.papermc.paper.inventory.item.properties.ItemProperties;
++import io.papermc.paper.property.PropertyHolder;
++import org.bukkit.DyeColor;
++import org.bukkit.Material;
++import org.bukkit.block.banner.Pattern;
++import org.bukkit.inventory.meta.BannerMeta;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++import java.util.List;
++
++public class MetaBanner extends PaperMeta implements BannerMeta {
```

Comment:
I don't like this naming scheme. I suggest sticking to what craftbukkit did, but with paper, so this would be `PaperMetaBanner`